src/evaluators/Program.js

```diff
@@ -246,6 +247,7 @@ export default function(ast: BabelNodeProgram, strictCode: boolean, env: Lexical
           emitConditionalThrow(res.joinCondition, res.consequent, res.alternate);
           res = res.value;
         } else if (res instanceof ThrowCompletion) {
+          issueThrowCompilerDiagnostic(res.value);
```

Comment:
This only happens if the program is known to always terminate with an exception. The error text suggests that this might not actually happen. Which is it?